### PR TITLE
Use axom::Path in IOManager

### PR DESCRIPTION
**Pull request by nselliott**
# Summary

- This PR is a small code cleanup
- It does the following:
  - Modifies some of the operations on path strings in IOManager to use axom::Path rather than conduit string utilities
  - Does not change the API
  - Addresses issue #754

### CODE REVIEW

File: src/axom/sidre/spio/IOManager.cpp
Comment (nselliott) [inline src/axom/sidre/spio/IOManager.cpp:806]: One limitation of axom::Path is that it does not preserve a leading '/' delimiter, which can make a meaningful difference when used as a file system path, so lines 803-806 restore this when necessary.

Comment (white238) [reply src/axom/sidre/spio/IOManager.cpp:806]: I would argue that's a bug and not intentional.  Did you attempt to fix this in `axom::Path` and other things broke?

Comment (nselliott) [reply src/axom/sidre/spio/IOManager.cpp:806]: I didn't touch `axom::Path`.  I could try to make a fix there before merging this PR.

Comment (nselliott) [reply src/axom/sidre/spio/IOManager.cpp:806]: It is fixable in `axom::Path` and I got most of the way there this afternoon.  However, I think it should be a standalone change and not bloat the scope of this PR.  I'm inclined to create another issue and PR.  I could merge this one with the 4-line workaround, or let it wait for the other one.

Comment (nselliott) [reply src/axom/sidre/spio/IOManager.cpp:806]: Issue #777 created regarding the leading delimiter in `axom::Path`.

Comment (white238) [reply src/axom/sidre/spio/IOManager.cpp:806]: i'm okay with that

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
Approved pending you fix your style problem.

File: src/axom/sidre/spio/IOManager.cpp
Comment (kennyweiss) [inline src/axom/sidre/spio/IOManager.cpp:803]: Suggestion: It might be cleaner to replace this with `axom::utilities::string::startsWith(root, delimiter[0])`

See: https://github.com/LLNL/axom/blob/7d595645e7e04e1f88c0f5b5cb265e7328eb0859/src/axom/core/utilities/StringUtilities.hpp#L43-L64

File: src/axom/sidre/spio/IOManager.cpp
Comment (kennyweiss) [reply src/axom/sidre/spio/IOManager.cpp:806]: I agree that this is a bug and that it can be resolved with the current workaround for this PR with a follow-up fix in `axom::Path`, as discussed in #777.

File: src/axom/sidre/spio/IOManager.cpp
Comment (kennyweiss) [inline src/axom/sidre/spio/IOManager.cpp:811]: This block could be replaced by:
```cpp
axom::utilities::string::appendPrefix(root_dir, file_name,  delimiter[0])
```
(which handles the check for empty).

See: https://github.com/LLNL/axom/blob/7d595645e7e04e1f88c0f5b5cb265e7328eb0859/src/axom/core/utilities/StringUtilities.hpp#L117-L130

**Review by kennyweiss — APPROVED** [review]
Thanks for working on this @nselliott 

I had a few suggestions for using more of the string utilities to clean the code up a bit more.

**Review by agcapps — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]
Thanks @nselliott

**Pull request merged**
